src/charm.py

```diff
@@ -1406,18 +1400,17 @@ def get_invalid_integration_status(self) -> Optional[StatusBase]:
                 "Relation to s3-integrator is not supported, config role must be config-server"
             )
 
-    def get_status(self) -> StatusBase:
-        """Returns the status with the highest priority from backups, sharding, and mongod.
-
-        Note: it will never be the case that shard_status and config_server_status are both present
-        since the mongodb app can either be a shard or a config server, but not both.
-        """
-        # retrieve statuses of different services running on Charmed MongoDB
+    def get_statuses(self) -> Tuple:
+        """Retrieves statuses for the different processes running inside the unit."""
         mongodb_status = build_unit_status(self.mongodb_config, self._unit_ip(self.unit))
         shard_status = self.shard.get_shard_status()
         config_server_status = self.config_server.get_config_server_status()
         pbm_status = self.backups.get_pbm_status()
+        return (mongodb_status, shard_status, config_server_status, pbm_status)
```

Comment:
For a future PR - this would be worth replacing by a SimpleNameSpace or a dict for clarity